File: mobile-apps/android/app/src/main/java/com/supermonng/mobile/domain/model/Node.kt
package com.supermonng.mobile.domain.model

import com.google.gson.annotations.SerializedName

data class Node(
    val id: String,
    @SerializedName("node_number")
    val nodeNumber: Int? = null,
    val callsign: String? = null,
    val description: String? = null,
    val location: String? = null,
    val status: NodeStatus? = null,
    @SerializedName("last_heard")
    val lastHeard: String? = null,
    @SerializedName("connected_nodes")
    val connectedNodes: List<ConnectedNode>? = null,
    @SerializedName("cos_keyed")
    val cosKeyed: Int? = null,
    @SerializedName("tx_keyed")
    val txKeyed: Int? = null,
    @SerializedName("cpu_temp")
    val cpuTemp: String? = null,
    @SerializedName("cpu_up")
    val cpuUp: String? = null,
    @SerializedName("cpu_load")
    val cpuLoad: String? = null,
    val alert: String? = null,
    val wx: String? = null,
    val disk: String? = null,
    @SerializedName("is_online")
    val isOnline: Boolean? = null,
    @SerializedName("is_keyed")
    val isKeyed: Boolean? = null,
    @SerializedName("created_at")
    val createdAt: String? = null,
    @SerializedName("updated_at")
    val updatedAt: String? = null,
    val info: String? = null,
    @SerializedName("remote_nodes")
    val remoteNodes: List<ConnectedNode>? = null,
    val ALERT: String? = null,
    val WX: String? = null,
    val DISK: String? = null
)

data class ConnectedNode(
    val node: String,
    val info: String,
    val ip: String? = null,
    @SerializedName("last_keyed")
    val lastKeyed: String,
    val link: String,
    val direction: String,
    val elapsed: String,
    val mode: String,
    val keyed: String
)

enum class NodeStatus {
    @SerializedName("online")
    ONLINE,
    @SerializedName("offline")
    OFFLINE,
    @SerializedName("connecting")
    CONNECTING,
    @SerializedName("error")
    ERROR,
    @SerializedName("unknown")
    UNKNOWN
}

enum class NodeActionType {
    @SerializedName("connect")
    CONNECT,
    @SerializedName("disconnect")
    DISCONNECT,
    @SerializedName("monitor")
    MONITOR,
    @SerializedName("local_monitor")
    LOCAL_MONITOR,
    @SerializedName("perm_connect")
    PERM_CONNECT,
    @SerializedName("reboot")
    REBOOT,
    @SerializedName("restart")
    RESTART
}

data class NodeAction(
    val type: NodeActionType,
    val nodeId: String,
    val targetNodeId: String? = null,
    val permanent: Boolean = false
)

File: frontend/src/types/node.ts
// Node and AllStar system types

export interface Node {
  id: string | number  // Allow both for compatibility
  node_number?: number
  callsign?: string
  description?: string
  location?: string
  status?: NodeStatus
  last_heard?: string
  connected_nodes?: ConnectedNode[]
  cos_keyed?: number
  tx_keyed?: number
  cpu_temp?: string
  cpu_up?: string
  cpu_load?: string
  alert?: string
  wx?: string
  disk?: string
  is_online?: boolean
  is_keyed?: boolean
  created_at?: string
  updated_at?: string
  // Legacy compatibility
  info?: string
  remote_nodes?: ConnectedNode[]
  // Additional AMI fields
  ALERT?: string
  WX?: string
  DISK?: string
}

export interface ConnectedNode {
  node: string
  info: string
  ip?: string
  last_keyed: string
  link: string
  direction: string
  elapsed: string
  mode: string
  keyed: string
}

export interface NodeConfig {
  [nodeId: string]: {
    host?: string
    port?: number
    hideNodeURL?: number
    lsnodes?: string
    listenlive?: string
    archive?: string
    menu?: string
    system?: string
    [key: string]: any
  }
}

export interface NodeStatistics {
  nodeId: string
  totalConnections: number
  activeConnections: number
  uptime: string
  lastActivity: string
  trafficStats: {
    rxBytes: number
    txBytes: number
    packets: number
  }
}

export interface NodeCommand {
  command: string
  nodeId: string
  parameters?: Record<string, any>
  timeout?: number
}

export interface AstDbEntry {
  [nodeId: string]: string[]
}

export interface NodeAction {
  type: NodeActionType
  nodeId: string
  targetNodeId?: string
  permanent?: boolean
}

// Node status types
export type NodeStatus = 
  | 'online'
  | 'offline'
  | 'connecting'
  | 'error'
  | 'unknown'

// Node action types
export type NodeActionType = 
  | 'connect'
  | 'disconnect'
  | 'monitor'
  | 'local_monitor'
  | 'perm_connect'
  | 'reboot'
  | 'restart'

// Node connection states
export type NodeConnectionState = 
  | 'idle'
  | 'receiving'
  | 'transmitting'
  | 'full_duplex'
  | 'cos'
  | 'ptt'

// Node types
export type NodeType = 
  | 'repeater'
  | 'simplex'
  | 'echolink'
  | 'irlp'
  | 'hub'
